do.go

```diff
@@ -835,6 +835,10 @@ func (d *DO) newResultSlicePointer() interface{} {
 	return reflect.New(reflect.SliceOf(reflect.PtrTo(d.modelType))).Interface()
 }
 
+func (d *DO) AddError(err error) error {
+	return d.withError(err).underlyingDB().Error
```

Comment:
maybe change the code to
```go
d.underlyingDB().AddError(err)
```?

Reply:
Thanks @jinzhu.

I have update the PR and noticed the old implementation won't propagate the error back correctly.